ir/visitor.h

```diff
@@ -65,8 +65,8 @@ struct Visitor_Context {
     }
 };
 
+class ControlFlowVisitor;
```

Comment:
Why not declare a new visitor class that is specialized to Loops instead of extending IR::Visitor? Seems like some dirty abstractions here. Are the extensions only relevant to ControlflowVisitors or do all visitors need to support them?

Reply:
ControlFlowVisitor is a mixin for doing control flow analysis over the IR, so it needs to be aware of all nodes that a relevant to control flow (loops and and otherwise).  This is a forward declaration so we can have a virtual method Visitor::controlFlowVisitor() that does a dynamic_cast<ControlFlowVisitor>, but more efficiently (Avoid the gcc/clang optimization bug that dynamic_cast is much slower than a virtual function call when it should not be).